crates/rsonpath-lib/src/query/parser.rs

```diff
@@ -374,6 +421,61 @@ mod tests {
         assert_eq!(result, Ok(("", LabelString::Owned("a".to_string()))));
     }
 
+    #[test]
+    fn nonnegative_array_index() {
+        let input = "[5]";
+
+        let result = super::array_index_selector()(input);
+
+        assert_eq!(result, Ok(("", 5.try_into().unwrap())));
+    }
+
+    #[test]
+    fn zero_array_index() {
+        let input = "[0]";
+
+        let result = super::array_index_selector()(input);
+
+        assert_eq!(result, Ok(("", 0.try_into().unwrap())));
+    }
+
+    #[test]
+    fn negative_array_index() {
+        let input = "[-5]";
+
+        super::array_index_selector()(input).unwrap_err();
+    }
+
+    #[test]
+    fn two_sixyfour_array_index() {
+        let input = "[18446744073709551616]";
+
+        super::array_index_selector()(input).unwrap_err();
+    }
+
+    #[test]
+    fn two_sixyfour_plus_one_array_index() {
+        let input = "[18446744073709551617]";
+
+        super::array_index_selector()(input).unwrap_err();
+    }
+
+    #[test]
+    fn two_pow_fiftythree_minus_one_array_index() {
+        let input = "[9007199254740991]";
+
+        let result = super::array_index_selector()(input);
+
+        assert_eq!(result, Ok(("", 9007199254740991.try_into().unwrap())));
```

Comment:
This is giving me a warning from clippy about long literals:

```suggestion
        assert_eq!(result, Ok(("", 9_007_199_254_740_991.try_into().unwrap())));
```